### PR TITLE
Add count to return object for GetDeviceType endpoint

### LINKED ISSUE #410: Add returning count to GetDeviceType endpoint
References user story #278

In the return object of the GetDeviceType endpoint, include the number of devices associated with them.

### PR DESCRIPTION
**Pull request by roguedeleter**
Closes #410

This PR implements including the Count in the return object for GetDeviceType. Basically, this count is referring to the number of Devices that are currently in the "devices" collection, with this specific DeviceTypeId being used.

### CODE REVIEW

File: BEIMA.Backend.Test/DeviceTypeFunctions/GetDeviceTypeTest.cs [hunk 1]
```diff
@@ -92,6 +93,9 @@ public void IdIsValid_GetDevice_ReturnsDevice()
             mockDb.Setup(mock => mock.GetDeviceType(It.Is<ObjectId>(oid => oid == new ObjectId(testId))))
                   .Returns(dbDeviceType.GetBsonDocument())
                   .Verifiable();
+            mockDb.Setup(mock => mock.GetFilteredDevices(It.IsAny<FilterDefinition<BsonDocument>>()))
```
Comment (josephmoore99) [inline BEIMA.Backend.Test/DeviceTypeFunctions/GetDeviceTypeTest.cs:96]: Let's add a mock verify for this

File: BEIMA.Backend/DeviceTypeFunctions/GetDeviceType.cs [hunk 2]
```diff
@@ -36,14 +36,20 @@ public static class GetDeviceType
...
             // Check that the device type returned is not null.
             if (doc is null)
             {
                 return new NotFoundObjectResult(Resources.DeviceTypeNotFoundMessage);
             }
 
+            // Find the total number of devices that are match this deviceType, and return the count of devices in the return object.
+            var filter = MongoFilterGenerator.GetEqualsFilter("deviceTypeId", doc.GetValue("_id"));
+            var deviceCount = mongo.GetFilteredDevices(filter).Count;
+            doc.Add("count", deviceCount);
```
Comment (josephmoore99) [inline BEIMA.Backend/DeviceTypeFunctions/GetDeviceType.cs:51]: Do we want this on get list as well?

Comment (roguedeleter) [reply BEIMA.Backend/DeviceTypeFunctions/GetDeviceType.cs:51]: That's a separate task #467

File: BEIMA.Backend.Test/DeviceTypeFunctions/GetDeviceTypeTest.cs (hunk 1 above)
Comment (roguedeleter) [reply BEIMA.Backend.Test/DeviceTypeFunctions/GetDeviceTypeTest.cs:96]: It's very difficult to verify the FilterDefinition, plus this is how we're already doing it in all the other unit tests. What should we verify, that it is not null?

Comment (josephmoore99) [reply BEIMA.Backend.Test/DeviceTypeFunctions/GetDeviceTypeTest.cs:96]: Assert.DoesNotThrow(mockDb.Verify(mock => mock.GetFilteredDevices(It.IsAny<FilterDefinition<BsonDocument>>()), Times.Once);

Comment (roguedeleter) [reply BEIMA.Backend.Test/DeviceTypeFunctions/GetDeviceTypeTest.cs:96]: I see, I thought you meant to verify the actual filter itself like on line 93. I will add this.

Comment (roguedeleter) [reply BEIMA.Backend.Test/DeviceTypeFunctions/GetDeviceTypeTest.cs:96]: fixed f3155ad

File: BEIMA.Backend.FT/DeviceTypeFT.cs [hunk 3]
```diff
@@ -85,6 +85,7 @@ public async Task NoDeviceTypesInDb_AddDeviceType_DeviceTypeAddedSuccessfullyAsy
             Assert.That(getDeviceType.Name, Is.EqualTo(device.Name));
             Assert.That(getDeviceType.Notes, Is.EqualTo(device.Notes));
             Assert.That(getDeviceType.Description, Is.EqualTo(device.Description));
+            Assert.That(getDeviceType.Count, Is.EqualTo(0));
```
[the remark this answers is not in the document]
Comment (roguedeleter) [reply BEIMA.Backend.FT/DeviceTypeFT.cs:88]: fixed 10cabf7

**Review by josephmoore99 — APPROVED** [review]

**Review by thessbsu — APPROVED** [review]

**Pull request merged**